src/Components/Subscriptions/index.js

```diff
@@ -1,9 +1,216 @@
+  const [button, setButton] = useState('');
+
+  const getSubscriptions = async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions`, {
+        method: 'GET'
+      });
+      const { data: subscriptions } = await response.json();
+      setSubscriptions(subscriptions);
+    } catch (error) {
+      alert(error.message);
+    }
+  };
+
+  const getSubscriptionsById = (id) => {
+    const subscription = subscriptions.find((subs) => subs._id === id);
+    if (subscription) {
+      setCreate({
+        _id: subscription._id ? subscription._id : '',
+        classes: subscription.classes ? subscription.classes._id : '',
+        member: subscription.member ? subscription.member._id : '',
+        date: subscription.date || ''
+      });
+      setShowForm(true);
+    }
+  };
+
+  const deleteSubscriptions = async (id) => {
+    try {
+      await fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions/${id}`, {
+        method: 'DELETE'
+      });
+      setSubscriptions((currentSubscriptions) => {
+        return currentSubscriptions.filter((subs) => subs._id !== id);
+      });
+      alert('Subscription successfully deleted.');
+    } catch (error) {
+      alert(error.message);
+    }
+  };
+
+  const onchangeInput = (e) => {
+    setCreate({
+      ...create,
+      [e.target.name]: e.target.value
+    });
+  };
+
+  const createSubscription = async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions/`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(create)
+      });
+      if (response.ok) {
+        const createdSubscription = await response.json();
+        setSubscriptions((currentSubscriptions) => {
+          return [...currentSubscriptions, createdSubscription];
+        });
+        alert('Subscription successfully created.');
+        setShowForm(false);
+      } else {
+        alert('Error creating the subscription.');
+      }
+    } catch (error) {
+      alert(error.message);
+    }
+  };
+
+  const updateSubscription = async (id) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/subscriptions/${id}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          classes: create.classes,
+          member: create.member,
+          date: create.date
+        })
+      });
+      if (response.ok) {
+        const updatedSubscription = await response.json();
+        setSubscriptions((currentSubscriptions) => {
+          return currentSubscriptions.map((subs) => {
+            if (subs._id === updatedSubscription._id) {
+              return updatedSubscription;
+            }
+            return subs;
+          });
+        });
+        setShowForm(false);
+        alert('Subscription updated correctly.');
```

Comment:
you should show the messages from the server

Reply:
done